lib/utils/application_insights.ts

```diff
@@ -0,0 +1,31 @@
+import * as ApplicationInsights from "applicationinsights";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { ServiceId } from "../api/definitions/ServiceId";
+
+export interface ITelemetryParams {
+  readonly operationId: NonEmptyString;
+  readonly operationParentId?: NonEmptyString;
+  readonly serviceId?: ServiceId;
+}
+
+export function getApplicationInsightsTelemetryClient(
```

Comment:
ApplicationInsights.TelemetryClient() is not thread safe if used with context.keys

Reply:
I don't understand this comment, can you elaborate? 

also, if this method returns a new instance on each call, it should be called `createApplicationInsightsTelemetryClient`

Reply:
I've put an explanation in the comments above the function. As the TelemetryClient can share some properties between track* calls, it is not safe to instantiate it at the module level (outside the index() functions) because you get a mutable shared state and results are messed up.

Reply:
the problem may be caused by mutating the `context` object before calling `track` - it looks like [this approach](https://github.com/Microsoft/ApplicationInsights-node.js/issues/387#issuecomment-383648111) may be better and makes it possible to share a single client

Reply:
the problem with that approach is that when you call 
`CorrelationContextManager.runWithContext(correlationContext, cb);`
you lose the return value of the wrapped callback.